### PR TITLE
Issue/21/refactor test

### PR DESCRIPTION
**Pull request by HuanDay**
#21

### CODE REVIEW

File: src/test.rs [hunk 1]
```diff
@@ -9,24 +9,41 @@ use pact::types::{Numeric, PactType, StringLike};
...
-    };
-    let mut methods: Vec<(String, Method)> = Default::default();
-    methods.push((method.name.clone(), method.clone()));
+fn init_method(name: &str, block_cooldown: Option<u32>, constraints: Option<Vec<u8>>) -> Method {
+    Method {
+        name: name.to_string(),
+        block_cooldown,
+        constraints,
+    }
+}
```
Comment (elim0322) [inline src/test.rs]: Could this be implemented as the constructor for `Method` in `src/lib.rs` like
```rust
impl Method {
    pub fn new() -> Self { ... }
    ...
}
```

File: src/test.rs [hunk 2]
```diff
@@ -9,24 +9,41 @@ use pact::types::{Numeric, PactType, StringLike};
...
 
-    let module = Module {
-        name: "module_test".to_string(),
-        block_cooldown: None,
+fn init_module(name: &str, block_cooldown: Option<u32>, methods: Vec<(String, Method)>) -> Module {
```
Comment (elim0322) [inline src/test.rs]: Similar to `Method` above, could this be implemented as a constructor for `Module`

Comment (HuanDay) [reply src/test.rs]: Good idea, will update about this

Comment (holygits) [reply src/test.rs]: +1. Have a look at this pattern also: https://doc.rust-lang.org/1.0.0/style/ownership/builders.html

File: src/test.rs [hunk 3]
```diff
@@ -6,27 +6,28 @@ use codec::{Decode, Encode};
...
 use std::vec::Vec;
 
-#[test]
-fn it_works_encode() {
-    let method = Method {
-        name: "method_test".to_string(),
-        block_cooldown: None,
-        constraints: None,
-    };
+fn config_methods(method: &Method) -> Vec<(String, Method)> {
```
Comment (holygits) [inline src/test.rs]: nitpick: can these be called `make_methods` or `build_methods` config doesn't seem right

File: src/test.rs [hunk 4]
```diff
@@ -6,27 +6,28 @@ use codec::{Decode, Encode};
...
+    methods
+}
 
-    let module = Module {
-        name: "module_test".to_string(),
-        block_cooldown: None,
-        methods,
-    };
-
+fn config_modules(module: &Module) -> Vec<(String, Module)> {
```
Comment (holygits) [inline src/test.rs]: nitpick: same here `make_modules` or `build_modules`

File: src/test.rs [hunk 5]
```diff
@@ -87,22 +77,13 @@ fn it_works_decode() {
...
-    let module = Module {
-        name: "module_test".to_string(),
-        block_cooldown: Some(86_400),
-        methods,
-    };
-
-    let mut modules: Vec<(String, Module)> = Default::default();
-    modules.push((module.name.clone(), module.clone()));
+    let module = Module::new("module_test")
+        .block_cooldown(86_400)
```
Comment (holygits) [inline src/test.rs]: nice one 👍

File: src/test.rs (hunk 3 above)
Comment (HuanDay) [reply src/test.rs]: No problem

**Pull request merged**